### DISCUSSION TITLE
Suggestion: Preview of the Next OTP for a Smoother Experience?

### DISCUSSION THREAD

**Discussion by shambu2k** [Authenticator-Extension/Authenticator#1056] [Ideas]
Hey there,

I've been using this extension for a while now, and it's a lifesaver. But, you know, there's this little thing that's been bugging me.

**Here's my thought:**

Right now, the extension gives me the current valid OTP, which is great, but these little devils change every 30 seconds or so. I've had moments where I'm just about to enter the code and poof—it changes. Frustrating, right?

So, how about a little tweak? Let's say we show the next valid OTP right under the current code, sort of like a sneak peek into the future. This way, if I see the timer ticking down on the current code, I could just use the next one and avoid the last-second scramble.

A rough idea of what it might look like:
<img width="317" alt="Screenshot 2023-05-16 at 11 12 51 PM" src="https://github.com/Authenticator-Extension/Authenticator/assets/49567934/bba55e98-09d3-491f-8c27-c7daf026b3e8">

**Why this could be cool:**

No more expired code drama: This could save users (like me) from the mini heart attack when an OTP expires just as we're about to enter it.
Stay ahead of the game: It'd be great to know the next OTP in line, especially if the current one's about to bow out.
And hey, not everyone might want this, so maybe we could let users toggle this feature on or off?

That's about it! I think this could be a neat addition and make the extension even more user-friendly. Thanks and keep up the good work!

**Comment by Sneezry** [comment]
Hi Sidharth, have you tried click the code to copy or the auto fill feature? I believe those features will save your time when typing codes.

**Reply by shambu2k** [reply]
> Hi Sidharth, have you tried click the code to copy or the auto fill feature? I believe those features will save your time when typing codes.

That works for sure but I have a few cases where I need to type the codes on a different device where this could really help.

**Reply by shambu2k** [reply]
Hey @Sneezry 
Got room to slot this in? We could have this as an optional feature with a toggle in the settings.
Also, thought I'd mention—I'm game to pitch in and get this feature rolling. Happy to code it up and send over a PR when it's ready.

Let me know what you think!

**Comment by madhu2000u** [comment]
Yes that's right! It's is absolutely frustrating sometimes and u get like a rush of adrenaline to race and enter the code before it expires. This would be a really cool extension.
